src/physics/src/spalart_allmaras_stab_helper.C

```diff
@@ -85,7 +85,13 @@ namespace GRINS
     libMesh::Number rhoUdotGradnu = rho*(U*grad_nu);
 
     // The diffusion term
-    libMesh::Number inv_sigmadivnuplusnuphysicalGradnu = (1./this->_sa_params.get_sigma())*(grad_nu*grad_nu + ((nu_value + mu)*(hess_nu(0,0) + hess_nu(1,1) + (this->_flow_vars.dim() == 3)?hess_nu(2,2):0)) + this->_sa_params.get_cb2()*grad_nu*grad_nu);
+    libMesh::Number inv_sigmadivnuplusnuphysicalGradnu = 1./this->_sa_params.get_sigma();
```

Comment:
What was the warning, and what's actually changed here?  I see what looks like one redundant parens set in the old code and a different redundant parens set in the new code; I don't see what I'd expect to evaluate differently or avoid a warning.

Reply:
<pre>
../../../pbauman/src/physics/src/spalart_allmaras_stab_helper.C:88:188: warning: operator '?:' has lower precedence than '+'; '+' will be evaluated first [-Wparentheses]
    libMesh::Number inv_sigmadivnuplusnuphysicalGradnu = (1./this->_sa_params.get_sigma())*(grad_nu*grad_nu + ((nu_value + mu)*(hess_nu(0,0) + hess_nu(1,1) + (this->_flow_vars.dim() == 3)?hess_nu(2,2):0)) + this->_sa_params.get_cb...
                                                                                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
../../../pbauman/src/physics/src/spalart_allmaras_stab_helper.C:88:188: note: place parentheses around the '+' expression to silence this warning
    libMesh::Number inv_sigmadivnuplusnuphysicalGradnu = (1./this->_sa_params.get_sigma())*(grad_nu*grad_nu + ((nu_value + mu)*(hess_nu(0,0) + hess_nu(1,1) + (this->_flow_vars.dim() == 3)?hess_nu(2,2):0)) + this->_sa_params.get_cb...
                                                                                                                                                                                           ^
                                                                                                                                (                                                          )
../../../pbauman/src/physics/src/spalart_allmaras_stab_helper.C:88:188: note: place parentheses around the '?:' expression to evaluate it first
    libMesh::Number inv_sigmadivnuplusnuphysicalGradnu = (1./this->_sa_params.get_sigma())*(grad_nu*grad_nu + ((nu_value + mu)*(hess_nu(0,0) + hess_nu(1,1) + (this->_flow_vars.dim() == 3)?hess_nu(2,2):0)) + this->_sa_params.get_cb...
</pre>

Reply:
So it seemed like from the warning that `hess_nu(1,1) + (this->_flow_vars.dim() == 3)` would be evaluated first and then `?hess_nu(2,2):0)` would be evaluated. So I put the parentheses around the conditional expression. The only test we have is 2d so we wouldn't have see this. I confess I've not bothered to double check the compiler on the order precedence, I just slapped in the parentheses to silence the warning because I could see that was the intention anyway.

Reply:
Oh no!  I forgot how low the ternary operator precedence was.  Good compiler!!